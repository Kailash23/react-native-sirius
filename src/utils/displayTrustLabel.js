/*
  Author: @jsworyk
  Description: Declare an array of trust labels that are acceptable for a given condition,
  if the array contains the trust label passed as an argument to the displayTrustLabel function return true and
  show the disclaimer label. Otherwise, the default response is false and the disclaimer is hidden.
*/
export const TRUST_LABEL_TYPES = {
  ADVERTISEMENT: 'advertisement',
  ADVICE: 'advice',
  ANALYSIS: 'analysis',
  EDITORIAL: 'editorial',
  EXPLAINER: 'explainer',
  FACT_CHECK: 'fact_check',
  FIRST_PERSON: 'first_person',
  INVESTIGATION: 'investigation',
  OPINION: 'opinion',
  REVIEW: 'review',
  SPONSORED: 'sponsored',
  SUPPORTED: 'supported'
};
// declare other groupings of labels here
let advertisementLabels = [
  TRUST_LABEL_TYPES.SPONSORED,
  TRUST_LABEL_TYPES.ADVERTISEMENT,
  TRUST_LABEL_TYPES.SUPPORTED
];
export const displayTrustLabel = (label, type) => {
  switch (type.toLowerCase()) {
    case TRUST_LABEL_TYPES.ADVERTISEMENT:
      if (advertisementLabels.includes(label.toLowerCase())) {
        return displayTrustLabel;
      }
      break;
  }
  return false;
};
