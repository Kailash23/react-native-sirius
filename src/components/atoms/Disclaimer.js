import React from 'react';
import { View, Platform } from 'react-native';
import { Text, withTheme } from 'react-native-paper';
import { SIZES, getFontSize } from '@styles/sizes';
import { customFonts } from '@styles/customFonts';
import { displayTrustLabel, TRUST_LABEL_TYPES } from '@utils/displayTrustLabel';

function Disclaimer(props) {
  const { data } = props;
  const { colors } = props.theme;
  return (
    <>
      {data &&
      data.trustlabel &&
      data.trustlabel.description &&
      displayTrustLabel(
        data.trustlabel.name,
        TRUST_LABEL_TYPES.ADVERTISEMENT
      ) ? (
        <View
          style={{
            borderColor: colors.onBackground,
            borderTopWidth: 0.5,
            borderBottomWidth: 0.5,
            paddingTop: SIZES.BASE,
            paddingBottom: SIZES.BASE,
            marginTop: SIZES.BASE,
            marginBottom: SIZES.BASE,
            marginLeft: SIZES.BASE,
            marginRight: SIZES.BASE,
            justifyContent: 'center',
            alignItems: 'center'
          }}>
          <Text
            style={{
              fontFamily: customFonts.merriweatherSansRegular,
              paddingTop: 0,
              paddingLeft: SIZES.BASE,
              paddingRight: SIZES.BASE,
              lineHeight: SIZES.BASE,
              borderColor: colors.onBackground,
              fontSize: getFontSize(SIZES.FONT),
              justifyContent: 'center',
              alignItems: 'center',
              color: colors.publishedDate
            }}>
            <Text
              style={{
                fontWeight: Platform.OS === 'ios' ? 'bold' : '800',
                fontFamily: customFonts.merriweatherSansBold,
                color: colors.disclaimerTitle
              }}>
              DISCLAIMER:{' '}
            </Text>
            {data.trustlabel.description}
          </Text>
        </View>
      ) : null}
    </>
  );
}

export default withTheme(Disclaimer);
