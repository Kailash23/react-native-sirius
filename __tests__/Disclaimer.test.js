import Disclaimer from '@atoms/Disclaimer';
import { displayTrustLabel, TRUST_LABEL_TYPES } from '@utils/displayTrustLabel';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
const moment = require('moment');

configure({ adapter: new Adapter() });

describe('Testing <Disclaimer/>', () => {
  test('Disclaimer should return nothing if the trustlabel is null', () => {
    const wrapper = shallow(<Disclaimer data={{ trustlabel: null }} />);
    expect(wrapper).toEqual({});
  });
  test('should render a Disclaimer with description', () => {
    const wrapper = shallow(
      <Disclaimer
        data={{
          trustlabel: { description: 'testDescription', name: 'testName' }
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  describe('displayTrustLabel(*) function Unit testing', () => {
    test('displayTrustLabel(*) should return false if trust lable is NOT sponsored / supported / advertisement ', () => {
      expect(
        displayTrustLabel('testName', TRUST_LABEL_TYPES.ADVERTISEMENT)
      ).toBe(false);
    });
  });
  describe('displayTrustLabel(*) function Unit testing', () => {
    test('displayTrustLabel(*) should return true if trust lable is sponsored / supported / advertisement ', () => {
      expect(
        displayTrustLabel('advertisement', TRUST_LABEL_TYPES.ADVERTISEMENT)
      ).toBeTruthy();
    });
  });
});
