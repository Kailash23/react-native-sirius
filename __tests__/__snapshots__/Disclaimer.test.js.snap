// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Disclaimer/> should render a Disclaimer with description 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  data={
    Object {
      "trustlabel": Object {
        "description": "testDescription",
        "name": "testName",
      },
    }
  }
/>
`;
